cores/esp8266/Schedule.cpp

```diff
@@ -76,90 +77,99 @@ bool schedule_function_us(std::function<bool(void)>&& fn, uint32_t repeat_us, sc
     return true;
 }
 
-IRAM_ATTR // (not only) called from ISR
-bool schedule_function_us(const std::function<bool(void)>& fn, uint32_t repeat_us, schedule_e policy)
+bool schedule_recurrent_function_us (const std::function<bool(void)>& fn, uint32_t repeat_us)
 {
-    return schedule_function_us(std::function<bool(void)>(fn), repeat_us, policy);
-}
+    assert(repeat_us < decltype(recurrent_fn_t::callNow)::neverExpires); //~26800000us (26.8s)
 
-IRAM_ATTR // called from ISR
-bool schedule_function(std::function<void(void)>&& fn, schedule_e policy)
-{
-    return schedule_function_us([fn]() { fn(); return false; }, 0, policy);
+    InterruptLock lockAllInterruptsInThisScope;
+
+    recurrent_fn_t* item = new recurrent_fn_t(repeat_us);
+    if (!item)
+        return false;
+
+    item->mFunc = fn;
+
+    if (rFirst)
+    {
+        item->mNext = rFirst;
+        rFirst = item;
+    }
+    else
+        rFirst = item;
+
+    return true;
 }
 
-IRAM_ATTR // called from ISR
-bool schedule_function(const std::function<void(void)>& fn, schedule_e policy)
+void run_scheduled_functions ()
 {
-    return schedule_function(std::function<void(void)>(fn), policy);
+    esp8266::polledTimeout::periodicFastMs yieldNow(100); // yield every 100ms
+
+    while (sFirst)
+    {
+        sFirst->mFunc();
+
+        {
+            InterruptLock lockAllInterruptsInThisScope;
+
+            auto to_recycle = sFirst;
+            sFirst = sFirst->mNext;
+            if (!sFirst)
+                sLast = nullptr;
+            recycle_fn_unsafe(to_recycle);
+        }
+
+        if (yieldNow)
+        {
+            // because scheduled function are allowed to last:
+            // this is yield() in cont stack:
+            esp_schedule();
+            cont_yield(g_pcont);
+        }
+    }
 }
 
-void run_scheduled_functions(schedule_e policy)
+void run_scheduled_recurrent_functions ()
 {
     // Note to the reader:
     // There is no exposed API to remove a scheduled function:
     // Scheduled functions are removed only from this function, and
     // its purpose is that it is never called from an interrupt
     // (always on cont stack).
 
+    if (!rFirst)
+        return;
+
     static bool fence = false;
     {
         InterruptLock lockAllInterruptsInThisScope;
```

Comment:
I'm not seeing the need for IRQ lock here.  this is a local static variable and the only assignments are below, no?